core/src/main/scala/kanaloa/reactive/dispatcher/queue/Worker.scala

```diff
@@ -138,15 +182,21 @@ trait Worker extends Actor with ActorLogging with MessageScheduler {
     }
   }
 
-  private def sendWorkToDelegatee(work: Work, retried: Int, delay: Option[FiniteDuration]): Unit = {
-    val timeoutHandle: Cancellable = delayedMsg(delay.fold(work.settings.timeout)(_ + work.settings.timeout), DelegateeTimeout)
-    maybeDelayedMsg(delay, work.messageToDelegatee, delegatee)
+  private def sendWorkToDelegatee(work: Work, retried: Int): Unit = {
+    val timeoutHandle: Cancellable = delayedMsg(delayBeforeNextWork.fold(work.settings.timeout)(_ + work.settings.timeout), RouteeTimeout)
+    delayBeforeNextWork match {
+      case Some(d) ⇒
+        import context.dispatcher
```

Comment:
I was more referring to the fact that this was duped, and we hadn't opened it up for having its own passed in ExecutionContext.  I'm all for having the ExecutionContext be specified per operation, for exactly the reasons you say(as well as not using the same context used by the Actor).  However, if we aren't going to that in this PR, I'd settle for at least passing in a context into the Actor, for use in these few cases.